### DISCUSSION TITLE
Text not rendered on the key / how import physical layout from a ZMK dtsi file?

### DISCUSSION THREAD

**Discussion by BenRoe** [caksoylar/keymap-drawer#141] [Q&A] [answered]
Hello,
some text is not positioned on the key. I tried to change some values in the RAW configuration, but it does not help.
[Link to the keymap](https://caksoylar.github.io/keymap-drawer?keymap_yaml=H4sIAAAAAAAC_92V207bQBCG73mKkZDwBUkMCeEQxEUOzqGYxE0ClAIK63hp3NjZ1AeiKPJr9G36Mn2SrnfBHlFTEK0q2pt_N994D7P772Qdmh1dh04XTOqwBdza1LFgYQcTIHBHnJCCY08prL6409GULk1GPKsCt9TzbF_1F5TOo7V1YB6smBdM2MghSxYGFVj5c8fmbeCFNAceW_gVKOVgzJzQnfF-OQfBJHRN3i1G8RQ-pTAJgrlfUdVPfP3QLIyZq47J1GdLh3gqX90l87zlkQX1VNNhpuoSe6YeaxcnVWM0MLR6wbXW5QbWkn381rb4MOr5lTUAk_g0bvPwXui50KZQQ2hN6DuhutBToRdCV3zRwxxMKjC4DTbhMBK0msT6IqbXh309SuBAwqo-TNlQstZpR7KW0JMk3n0Uj5mWMU_n8YI9oR-FfhBaF9oQeib0WGg7mUTJKWIaRSSVU9LpC2muhZSqKVVTWhsYdRHwl24O-NFXHf5BTNNv9EG7KVP3A3s8XYrvlI0xmfujBfMs5VWfDoxqXR7OjNzFmDeVJPr96zduh-Wcxn4hMz_6qxGj2tLgVHqrHjibYLQefj6ija68KE3Xhtozs4oLlz1uB9nhXnnJXhq9c7mQrjXlyIT0O622RFdv5PSejrR7J_KQtG4jPUbhhKyf3JMy6dhVzy4qyITX0BjwoamZlG0lNf92ukmliHgR8RLiJcR3EN9BvIx4GfFdxHcR30N8D_F9xPcRP0D8APEtxLcQv0H8Br3O2IDodaIIN2R2gBs0MyCu6slhV-n6V2hfebSvPOJHiB8hfolL3CUqHso1jlwr0Zu3_p-NaN2h1s_y_S_rT_KMeKnlLbE-p2-kJi-SWyr6iW1nsGIGK92z2hDqeh-quqx2A13TjFenWut3Hqps3L2vv8qGH4SWzUbhzGHjqfIP3-VZT39IMO4mCZqMBQ4jFvWU_9LDWRm-2Msv_CP4AQb6FeVkCwAA)

How can i import the physical layout from a [ZMK desi file](https://github.com/carrefinho/forager-zmk-module/blob/main/boards/shields/forager/forager.dtsi)?

**Comment by caksoylar** [comment]
> some text is not positioned on the key. I tried to change some values in the RAW configuration, but it does not help.

Hey, nothing jumps out to me here, which key do you mean that is not positioned well?

> How can i import the physical layout from a [ZMK desi file](https://github.com/carrefinho/forager-zmk-module/blob/main/boards/shields/forager/forager.dtsi)?

In local use, you can pass it via the [`-d/--dts-layout` arg](https://github.com/caksoylar/keymap-drawer/blob/main/KEYMAP_SPEC.md#zmk-physical-layout-specification), like
```sh
keymap parse -z boards/shields/forager/forager.keymap | keymap draw -d boards/shields/forager/forager.dtsi - >forager.svg
```

The web app doesn't let you upload a dtsi like that, I was planning to add support for it, as an additional option in this menu:
![image](https://github.com/user-attachments/assets/f8c44372-b379-4d63-b89a-008610fa1e64)

I'll see if I can quickly add that.

**Reply by caksoylar** [reply, marked as answer]
The web app now supports uploading dtsi to Layout override, so please try that with https://github.com/carrefinho/forager-zmk-module/blob/main/boards/shields/forager/forager.dtsi.

**Reply by BenRoe** [reply]
Wow that was fast. Many thanks for this great tool. 🙏

**Reply by BenRoe** [reply]
In the PNG the text is within the keycap, but in the browser not. Maybe a browser format thing.
Safari 18.1

<img width="761" alt="Bildschirmfoto 2024-12-14 um 15 57 54" src="https://github.com/user-attachments/assets/7a9c188c-2990-4ad8-8bcf-4a780574f938" />

**Reply by caksoylar** [reply]
Ah, yes, I can reproduce on my phone with Safari renderer. It is using [`dominant-baseline: hanging;`](https://github.com/caksoylar/keymap-drawer/blob/c5175d5a11a04277d2b2cd9b08355e33d032de10/keymap_drawer/config.py#L174) which seems to be ignored, but weirdly I can't reproduce it with a simple SVG example in a playground. I'll look into it a bit more.
